integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSource.scala

```diff
@@ -233,10 +233,17 @@ object CarbonSource {
    * @return
    */
   def updateCatalogTableWithCarbonSchema(tableDesc: CatalogTable,
-      sparkSession: SparkSession): CatalogTable = {
+                                         sparkSession: SparkSession): CatalogTable = {
     val metaStore = CarbonEnv.getInstance(sparkSession).carbonMetastore
     val storageFormat = tableDesc.storage
     val properties = storageFormat.properties
+    tableDesc.storage.properties.get("tableName") match {
```

Comment:
is it `tableName` mandatory here? tableName validation is not happened in parser?

Reply:
@ravipesala 1.tableName validation does not take place in parser when you datasource ddl
2.test case is already there in carbondatasource suite i just modified it to check that when exception is thrown table should not be there in store